### ISSUE TITLE
Add rhub workflow?

### ISSUE THREAD

**Issue by rempsyc** [easystats/easystats#426]
When checking packages before a CRAN submission, using `devtools::release()`, we get,

```
Have you checked on R-hub (with `check_rhub()`)?

1: Definitely
2: No way
3: Nope
```

But now when using `rhub::check_for_cran()`, we get,

``` r
packageVersion("rhub")
#> [1] '2.0.0'
```

``` r
rhub::check_for_cran()
#> This function is deprecated and defunct since rhub v2.
#> Please see `?rhubv2` on transitioning to the new rhub functions.
```

<sup>Created on 2024-07-08 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

The docs mention,

> R-hub v2, i.e. version 2 or later of the rhub package is a completely new check system. In this acticle we highlight the differences between the old and the new system.
> 
> There are two ways to use R-hub v2. The recommended way is to store your R package in a GitHub repository and use the `⁠rhub_*()⁠` functions to start checks on GitHub Actions, using your own GitHub account.
> 
> Call `rhub_setup()` from the local git clone to set up R-hub v2 for your package. This adds a GitHub Actions workflow to your local repository. Push this change to GitHub, into your default git branch and then you are ready to call start checks with `rhub_check()`.

As I am preparing to resubmit `report()`, I was going to add this workflow to the repo I maitain, but wonder whether it would make sense to add this workflow to all easystats repos. Note that this workflow isn't triggered automatically, it has to be explicitely called with `rhub_check()`, but it still requires the workflow on the repo.

**Comment by IndrajeetPatil** [comment]
Yes, we were planning to do this eventually: https://github.com/easystats/workflows/issues/36
